### PR TITLE
Strip spaces from incoming form data

### LINKED ISSUE #2828: Exception is raised when a source metadata identifier has an extra space
Should we trim any accidental spaces?

## Backtrace

    line 44 of [PROJECT_ROOT]/app/services/pul_metadata_services/client.rb: retrieve_from_pulfa
    line 14 of [PROJECT_ROOT]/app/services/pul_metadata_services/client.rb: retrieve
    line 44 of [PROJECT_ROOT]/app/models/remote_record.rb: client_result

[View full backtrace and more info at honeybadger.io](https://app.honeybadger.io/projects/53391/faults/47767616)

```
URI::InvalidURIError: bad URI(is not URI?): CD-18264 .xml

https://figgy.princeton.edu/concern/scanned_resources/new/recording
```

https://github.com/pulibrary/figgy/blob/master/app/services/pul_metadata_services/client.rb#L44

### PR DESCRIPTION
**Pull request by hackmastera**
closes #2828

### CODE REVIEW

File: spec/resources/scanned_resources/scanned_resources_controller_spec.rb [hunk 1]
```diff
@@ -201,6 +201,21 @@
       end
     end
 
+    context "when an attribute has leading / trailig spaces" do
```
Comment (escowles) [inline spec/resources/scanned_resources/scanned_resources_controller_spec.rb]: Typo: "trailig"

File: spec/resources/scanned_resources/scanned_resources_controller_spec.rb [hunk 2]
```diff
@@ -201,6 +201,21 @@
...
+      let(:resource) { FactoryBot.create_for_repository(:scanned_resource) }
+      let(:params) do
+        {
+          source_metadata_identifier: " CD-18264 "
```
Comment (escowles) [inline spec/resources/scanned_resources/scanned_resources_controller_spec.rb]: I'm 👍 on the implementation, but not sure about using an invalid source metadata identifier in this test.

Comment (hackmastera) [reply spec/resources/scanned_resources/scanned_resources_controller_spec.rb]: hm, good point. I was using the data from the error, but it's incidental that it was bad. I'll sub in something valid.

**Comment by hackmastera** [comment]
@escowles I made the changes you requested and there were also some broken specs which required updates to the implementation. If you could re-review!

**Pull request merged**